domparsing/DOMParser-parseFromString-xml-internal-subset.html

```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<title>Parsing and serialization of doctype internal subset</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+test(function () {
+  // https://www.w3.org/TR/xml11/#sec-prolog-dtd has intSubset as part of the
+  // syntax, which is not represented in the DocumentType interface. Check that
+  // publicId and systemId are not affected.
+  var doc = new DOMParser().parseFromString('<!DOCTYPE foo [ <!ELEMENT foo (#PCDATA)> ]><foo/>', 'text/xml');
+  var doctype = doc.doctype;
+  assert_equals(doctype.name, 'foo', 'doctype name');
+  assert_equals(doctype.publicId, '', 'doctype publicId');
+  assert_equals(doctype.systemId, '', 'doctype systemId');
+  // Sanity check: the root element is still parsed correctly.
+  var documentElement = doc.documentElement;
+  assert_equals(documentElement.localName, 'foo', "root element's localName");
+  // https://w3c.github.io/DOM-Parsing/#xml-serializing-a-documenttype-node also
+  // does not have any notion of the internal subset, so also check that it
+  // isn't serialized by XMLSerializer.
```

Comment:
This does match the spec, but do we really want the spec to say this?  What's the rationale for not round-tripping here?  I would think the spec should be changed to match Firefox and Edge.  If for some reason we really want browsers to throw away the internal subset when reserializing and change the semantics of the XML document, possibly converting it from valid to invalid XML, then LGTM.  But I'd like to hear why.  :)